ksqldb-rest-app/src/main/java/io/confluent/ksql/api/endpoints/KsqlServerEndpoints.java

```diff
@@ -42,17 +46,22 @@
   private final PullQueryExecutor pullQueryExecutor;
   private final ReservedInternalTopics reservedInternalTopics;
   private final KsqlSecurityContextProvider ksqlSecurityContextProvider;
+  private final KsqlStatementsEndpoint ksqlStatementsEndpoint;
+  private final TerminateEndpoint terminateEndpoint;
 
   public KsqlServerEndpoints(
       final KsqlEngine ksqlEngine,
       final KsqlConfig ksqlConfig,
       final PullQueryExecutor pullQueryExecutor,
-      final KsqlSecurityContextProvider ksqlSecurityContextProvider) {
+      final KsqlSecurityContextProvider ksqlSecurityContextProvider,
+      final KsqlResource ksqlResource) {
     this.ksqlEngine = Objects.requireNonNull(ksqlEngine);
     this.ksqlConfig = Objects.requireNonNull(ksqlConfig);
     this.pullQueryExecutor = Objects.requireNonNull(pullQueryExecutor);
     this.reservedInternalTopics = new ReservedInternalTopics(ksqlConfig);
     this.ksqlSecurityContextProvider = Objects.requireNonNull(ksqlSecurityContextProvider);
+    this.ksqlStatementsEndpoint = new KsqlStatementsEndpoint(Objects.requireNonNull(ksqlResource));
```

Comment:
nit: no need to null check `ksqlResource` here - that should be the responsibility of the `KsqlStatementsEndpoint` constructor...

Reply:
ack